community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_3/runtime/pipes/PruningVarLengthExpandPipe.scala

```diff
@@ -260,18 +261,22 @@ case class PruningVarLengthExpandPipe(source: Pipe,
         (Empty, null)
       } else {
         val row = input.next()
-        val nextState = new PrePruningDFS(whenEmptied = this,
-                                          node = getNodeFromRow(row),
-                                          path = new Array[Long](max),
-                                          pathLength = 0,
-                                          state = state,
-                                          row = row,
-                                          expandMap = Primitive.longObjectMap[FullExpandDepths]())
-        nextState.next()
+        row.get(fromName) match {
+          case Some(node: NodeValue) =>
+            val nextState = new PrePruningDFS(whenEmptied = this,
+              node = node,
+              path = new Array[Long](max),
+              pathLength = 0,
+              state = state,
+              row = row,
+              expandMap = Primitive.longObjectMap[FullExpandDepths]())
+            nextState.next()
+          case Some(x: Value) if x.asObject() == null =>
```

Comment:
This is correct but slightly strange. I would prefer 
`case Some(x: Value) if x == NO_VALUE =>`
or even
`case Some(NO_VALUE) =>`
if that's allowed

Reply:
We tried to write i prettier but we didn't managed, must ave been trying to use the wrong NO_VALUE or something. However, fixed it now in a new commit.